pkg/controller/statusmanager/pod_status.go

```diff
@@ -72,24 +73,36 @@ type statefulsetState struct {
 func (status *StatusManager) SetFromPods() {
 	status.Lock()
 	defer status.Unlock()
+	selector, err := labels.Parse(generateStatusSelector)
+	if err != nil {
+		panic(err) // selector is guaranteed valid, unreachable
+	}
+
+	daemonSets, _ := status.dsLister.List(selector) // informer, cannot fail
+	deployments, _ := status.depLister.List(selector)
+	statefulSets := []*appsv1.StatefulSet{}
+	for _, lister := range status.ssListers {
+		l, _ := lister.List(selector)
+		statefulSets = append(statefulSets, l...)
+	}
 
 	targetLevel := os.Getenv("RELEASE_VERSION")
-	reachedAvailableLevel := (len(status.daemonSets) + len(status.deployments) + len(status.statefulSets)) > 0
+	reachedAvailableLevel := (len(daemonSets) + len(deployments) + len(statefulSets)) > 0
 
 	progressing := []string{}
 	hung := []string{}
 
 	daemonsetStates, deploymentStates, statefulsetStates := status.getLastPodState()
 
-	for _, dsName := range status.daemonSets {
-		ds := &appsv1.DaemonSet{}
-		if err := status.client.ClientFor(dsName.ClusterName).CRClient().Get(context.TODO(), types.NamespacedName{Namespace: dsName.Namespace, Name: dsName.Name}, ds); err != nil {
-			log.Printf("Error getting DaemonSet %q: %v", dsName.String(), err)
-			progressing = append(progressing, fmt.Sprintf("Waiting for DaemonSet %q to be created", dsName.String()))
-			reachedAvailableLevel = false
-			// Assume the OperConfig Controller is in the process of reconciling
-			// things; it will set a Degraded status if it fails.
-			continue
+	if (len(daemonSets) + len(deployments) + len(statefulSets)) == 0 {
+		progressing = append(progressing, "Deploying")
+	}
+
+	for _, ds := range daemonSets {
+		dsName := ClusteredName{
+			Namespace:   ds.Namespace,
+			Name:        ds.Name,
+			ClusterName: apply.GetClusterName(ds),
```

Comment:
though not really related to this patch:
1. It is not clear why `GetClusterName` should be in `pkg/apply`
2. `dsName := NewClusteredName(ds)`
```
func NewClusteredName(obj metav1.Object) ClusteredName {
        return ClusteredName{
                Namespace:   obj.GetNamespace(),
                Name:        obj.GetName(),
                ClusterName: obj.GetAnnotations()[name.ClusterNameAnnotation],
        }
}
```

Reply:
1. Agreed, but changing that is another PR.
2. That is a good idea.